tools/build-tool/code/tooldebian.pas

```diff
@@ -0,0 +1,192 @@
+  begin
+    case Cpu of
+      { Architectures supported by https://packages.debian.org/buster/bash }
+      x86_64: Result := 'amd64';
+      aarch64: Result := 'arm64';
+      arm: begin
+          Result := 'armel';
+          WriteLn('WARNING: Architecture ' + CpuToString(Cpu) + ' is ambiguous between "armel" and "armhf" in Debian.');
+        end;
+      i386: Result := 'i386';
+      mips: Result := 'mips';
+      mipsel: Result := 'mipsel';
+      powerpc64: begin
+          Result := 'ppc64el';
+          WriteLn('WARNING: Architecture ' + CpuToString(Cpu) + ' is ambiguous between "ppc64el" and "ppc64" in Debian.');
+        end;
+      // "mips64el"
+      // "s390x"
+      { Unofficial Debian ports https://packages.debian.org/sid/bash }
+      m68k: Result := 'm68k';
+      riscv64: Result := 'riscv64';
+      sparc64: Result := 'sparc64';
+      // "alpha"
+      // "sh4"
+      // "x32" = X32 is an ABI for amd64/x86_64 CPUs using 32-bit integers, longs and pointers
+      else {powerpc,sparc,avr,jvm,i8086,riscv32,armeb}
+      begin
+        Result := CpuToString(Cpu);
+        WriteLn('WARNING: Architecture ' + CpuToString(Cpu) + ' is not officially supported by Debian.');
+      end;
+    end;
+  end;
+
+var
+  PathToExecutableLocal, PathToExecutableUnix: String;
+  PathToIconFileLocal, PathToIconFileUnix: String;
+  ShareDir: String;
+  TextWriter: TTextWriter;
+  OutString: String;
+  PackageFolder: String;
+begin
+  PackageFolder := 'castle-engine-output' + PathDelim + PackageFileName;
+  if DirectoryExists(PackageFolder) then
+    RemoveNonEmptyDir(PackageFolder);
+
+  // Copy the binaries
+
+  PathToExecutableUnix := '/usr/' + Manifest.DebianInstallFolder + '/' + Manifest.Name;
+  PathToExecutableLocal := StringReplace(PathToExecutableUnix, '/', PathDelim, [rfReplaceAll]);
+  CopyDirectory(Path, PackageFolder + PathToExecutableLocal);
+  ShareDir := PackageFolder + PathDelim + 'usr' + PathDelim + 'share';
+  CreateDir(ShareDir);
+
+  // Calculate binaries size
+
+  BinariesSize := 0;
+  FindFiles(PackageFolder + PathToExecutableLocal, '*', false, {$ifdef CASTLE_OBJFPC}@{$endif} FoundFile, [ffRecursive]);
+
+  // Copy XPM icon
+
+  CreateDir(ShareDir + PathDelim + 'pixmaps');
+  PathToIconFileUnix := '/usr/share/pixmaps/' + Manifest.ExecutableName + '.xpm';
+  PathToIconFileLocal := StringReplace(PathToIconFileUnix, '/', PathDelim, [rfReplaceAll]);
+  if Manifest.Icons.FindExtension(['.xpm']) <> '' then
+    CheckCopyFile(Manifest.Icons.FindExtension(['.xpm']), PackageFolder + PathToIconFileLocal)
+  else
+  begin
+    // using ImageMagic - FPWriteXPM first doesn't properly write alpha channel, second uses palette char size = 2 which is not a good idea for an icon
+    //RunCommandSimple(FindExe('convert'), [Manifest.Icons.FindExtension(['.png']), PackageFolder + PathToIconFileLocal]);
+    if not RunCommand('/bin/convert', [Manifest.Icons.FindExtension(['.png']), PackageFolder + PathToIconFileLocal], OutString) then
```

Comment:
Better to not specify absolute path `/bin/convert`.

Reason: Is it `/bin/convert` or `/usr/bin/convert`? On my Debian I have both, because on latest Debian `/bin` is just a symlink to `/usr/bin`. But that is not true on all Unix systems, and not on older Debian versions. `/usr/bin` is actually a traditional place for non-critical programs, like ImageMagick I guess. Better to search on `$PATH`.

To do this, use `FindExe`:

```
ConvertExe := FindExe('convert');
if ConvertExe = '' then 
  raise Exception.Create('Cannot find "convert", install ImageMagick or GraphicsMagick');
if not RunCommand(ConvertExe, ...
```

Bonus gain is that it makes the code more portable. In the (distant probably...) possibility that e.g. Cygwin will one day package Debian tools, maybe this can work on Windows? Above `FindExe` approach would work OK.

Reply:
Moreover, in this case, as you don't really want to catch `OutString`, just use `RunCommandSimple`.

And it will automatically search `$PATH`. So just

```
RunCommandSimple('convert', [Manifest.Icons.FindExtension(['.png']), ...
```

Reply:
Many thanks for all the suggestions! Should be easy to fix. For now just a note here:

```
Manifest.Icons.FindExtension(['.png'])
```

Literally finds only "png" icon file. I wasn't sure how to proceed here. We need some "main" icon file, but most certainly not "ico" which produces broken results when converted (all icon sizes dumped into one image). But this icon may be any CGE-readable.

I thought maybe it's even a good idea to write our own XPM exporter for this purpose. It's really straightforward and in case of an app icon can be simplified even further. It won't be universal, as will work reliably only for small images, but specifically for this purpose.

Reply:
Fixed + improved a bit. Still the question about png...

Reply:
For now, I'd say it's OK to require just PNG. Indeed we need anything that is readable by ImageMagick, so in general this could be better, but for now I suppose just requiring PNG is cool.